interfaces/connection.go

```diff
@@ -0,0 +1,245 @@
+func (plug *ConnectedPlug) Hooks() map[string]*snap.HookInfo {
+	return plug.plugInfo.Hooks
+}
+
+// SecurityTags returns the security tags for this plug.
+func (plug *ConnectedPlug) SecurityTags() []string {
+	return plug.plugInfo.SecurityTags()
+}
+
+// StaticAttr returns a static attribute with the given key, or error if attribute doesn't exist.
+func (plug *ConnectedPlug) StaticAttr(key string) (interface{}, error) {
+	if val, ok := plug.staticAttrs[key]; ok {
+		return val, nil
+	}
+	return nil, fmt.Errorf("attribute %q not found", key)
+}
+
+// StaticAttrs returns all static attributes.
+func (plug *ConnectedPlug) StaticAttrs() map[string]interface{} {
+	return plug.staticAttrs
+}
+
+// Attr returns a dynamic attribute with the given name. It falls back to returning static
+// attribute if dynamic one doesn't exist. Error is returned if neither dynamic nor static
+// attribute exist.
+func (plug *ConnectedPlug) Attr(key string) (interface{}, error) {
+	if plug.dynamicAttrs != nil {
+		if val, ok := plug.dynamicAttrs[key]; ok {
+			return val, nil
+		}
+	}
+	return plug.StaticAttr(key)
+}
+
+// Attrs returns all dynamic attributes of this plug.
+func (plug *ConnectedPlug) Attrs() map[string]interface{} {
+	if plug.dynamicAttrs == nil {
+		plug.dynamicAttrs = make(map[string]interface{})
+	}
+	return plug.dynamicAttrs
+}
+
+// SetAttr sets the given dynamic attribute. Error is returned if the key is already used by a static attribute.
+func (plug *ConnectedPlug) SetAttr(key string, value interface{}) error {
+	if _, ok := plug.staticAttrs[key]; ok {
+		return fmt.Errorf("cannot change attribute %q as it was statically specified in the snap details", key)
+	}
+	if plug.dynamicAttrs == nil {
+		plug.dynamicAttrs = make(map[string]interface{})
+	}
+	// make a normalized copy
+	plug.dynamicAttrs[key] = copyRecursive(value)
+	return nil
+}
+
+// Interface returns the name of the interface for this slot.
+func (slot *ConnectedSlot) Interface() string {
+	return slot.slotInfo.Interface
+}
+
+// Name returns the name of this slot.
+func (slot *ConnectedSlot) Name() string {
+	return slot.slotInfo.Name
+}
+
+// Snap returns the snap Info of this slot.
+func (slot *ConnectedSlot) Snap() *snap.Info {
+	return slot.slotInfo.Snap
+}
+
+// Apps returns all the apps associated with this slot.
+func (slot *ConnectedSlot) Apps() map[string]*snap.AppInfo {
+	return slot.slotInfo.Apps
+}
+
+// SecurityTags returns the security tags for this slot.
+func (slot *ConnectedSlot) SecurityTags() []string {
+	return slot.slotInfo.SecurityTags()
+}
+
+// StaticAttr returns a static attribute with the given key, or error if attribute doesn't exist.
+func (slot *ConnectedSlot) StaticAttr(key string) (interface{}, error) {
+	if val, ok := slot.staticAttrs[key]; ok {
+		return val, nil
+	}
+	return nil, fmt.Errorf("attribute %q not found", key)
+}
+
+// StaticAttrs returns all static attributes.
+func (slot *ConnectedSlot) StaticAttrs() map[string]interface{} {
```

Comment:
same question

Reply:
I guess you meant Attrs() for slot here? See above.